cmd/gcp-controller-manager/node_csr_approver_test.go

```diff
@@ -1216,6 +1216,24 @@ func TestShouldDeleteNode(t *testing.T) {
 			getInstanceErr: testErr,
 			expectedErr:    testErr,
 		},
+		{
+			desc: "node with different instance id",
```

Comment:
Hi @bobbypage, I added a test case here. Let me know if more test cases we need.

Reply:
Looks great, thanks for adding it!